### PR TITLE
Fishtank fixes

### LINKED ISSUE #17130: Fish egg scoop fails to scoop up eggs and wall mounted fish tanks are a black hole for water.
Using the fish egg scoop up eggs out of the fish tank right now gives you nothing not sure if it null spaces the eggs you do scoop fails to  put them in your hand or on the floor.

wall mounted fish tank takes TOO long to fill up takes 6000 units of water to get it half filled right now.

### LINKED ISSUE #17143: bugged fish egg scoop
(WEB REPORT BY: davidsamathy REMOTE: 172.93.109.202:7777)
# Revision
61ac09958eace14213af691458932432085fbeb0

# Description
multiple fish of the same type produce eggs. the fish egg scoop removes them however does not create egg item for fish production
# Steps to Reproduce
build aquarium. place two fish of the same type inside. care for food and cleanliness until description of fish tank states fish tank has eggs. use egg scoop on fish tank
# What you Expected
egg clutches such as what are in the cargo crates to be produced upon use of the egg scoop
# What Actually Happened
the egg scoop removes eggs from description but no egg item(s) are created

### PR DESCRIPTION
**Pull request by ShiftyRail**
Closes #17130
Closes #17143

### CODE REVIEW

File: code/modules/fish/fish_tank.dm [hunk 1]
```diff
@@ -165,7 +165,7 @@
 	overlays.Cut()
 	//Update Alert Lights
 	if(has_lid)											//Skip the alert lights for aquariums that don't have lids (fishbowls)
-		if(egg_list.len)								//There is at least 1 egg to harvest
+		if(egg_list.len)								//There is at least 1 egg to
```
Comment (DamianX) [inline code/modules/fish/fish_tank.dm]: ?

Comment (ShiftyRail) [reply code/modules/fish/fish_tank.dm]: I can explain, it's to

File: code/modules/fish/fish_tank.dm [hunk 2]
```diff
@@ -350,23 +351,18 @@
...
-	for(var/i in egg_list.len)						//Loop until you've harvested all the eggs
-		var/obj/item/fish_eggs/egg = pick(egg_list)	//Select an egg at random
-		egg = new egg(get_turf(user))				//Spawn the egg at the user's feet
-		egg_list.Remove(egg)						//Remove the egg from the egg_list
+	for(var/i = 1 to egg_list.len)						//Loop until you've harvested all the eggs
+		var/obj/item/fish_eggs/egg = egg_list[i]	//Go through the eggs
+		new egg(get_turf(user))						//Spawn the egg at the user's feet
 
-	egg_list.Cut()									//Destroy any excess eggs, clearing the egg_list
+	egg_list = list()								//Destroy any excess eggs, clearing the egg_list
```
Comment (DamianX) [inline code/modules/fish/fish_tank.dm:365]: why this?

Comment (ShiftyRail) [reply code/modules/fish/fish_tank.dm:365]: I wanted to explicitly clean the list, but Cut() does the same thing so yeah
If Cut() is better, then I'll change it back

Comment (DamianX) [reply code/modules/fish/fish_tank.dm:365]: image

Comment (gbasood) [reply code/modules/fish/fish_tank.dm:365]: list.len = 0 too
they're all basically the same

**Pull request merged**